cpp/src/reshape/byte_cast.cu

```diff
@@ -135,9 +135,14 @@ struct byte_list_conversion_fn<T, std::enable_if_t<std::is_same_v<T, cudf::strin
         input.size(), output_type, stream, mr);
     }
 
-    auto col_content     = std::make_unique<column>(input, stream, mr)->release();
-    auto const num_chars = col_content.data->size();
-    auto uint8_col       = std::make_unique<column>(
+    auto const num_chars = strings_column_view(input).chars_size(stream);
+    CUDF_EXPECTS(num_chars < static_cast<int64_t>(std::numeric_limits<size_type>::max()),
```

Comment:
Why is this one a cast to `int64_t` and the other a cast to `size_t`?

Reply:
Because the type of `num_chars` here is `int64_t`

Reply:
Forgive my ignorance, but I've lost track of why `chars_size` returns an `int64_t` when the number of characters in a `string` column can't exceed `size_type` limits. Wasn't this changed to `int64_t` to allow for super-large string columns?

Reply:
Yes, the chars limit to `size_type` has been removed though this behavior is runtime dependent on an environment variable so as to not break upstream libraries (e.g. Spark).
This has been an ongoing project of work mainly documented here: https://github.com/rapidsai/cudf/issues/13733